sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileSystems.java

```diff
@@ -567,14 +568,17 @@ static FileSystem getFileSystemInternal(String scheme) {
    *
    * <p>Outside of workers where Beam FileSystem API is used (e.g. test methods, user code executed
    * during pipeline submission), consider use {@link #registerFileSystemsOnce} if initialize
-   * FIleSystem of supported schema is the main goal.
+   * FileSystem of supported schema is the main goal.
    */
   @Internal
   public static void setDefaultPipelineOptions(PipelineOptions options) {
     checkNotNull(options, "options cannot be null");
     long id = options.getOptionsId();
     int nextRevision = options.revision();
 
+    // entry to set other PipelineOption determined flags
+    Metrics.setDefaultPipelineOptions(options);
```

Comment:
This makes this a little overloaded no? seems odd we set metric stuff in FS. But I see you want to do this in centralized place. 
Would it better to have it separate in DataflowBatchWorkerHarness.java, StreamingDataflowWorker.java and FnHarness.java from where all this is called? 

(I do not have a preference, just a though. Feel free to ignore this comment if you want to do it this way)

Reply:
setDefaultPipelineOptions will be called once per worker, on SDK harness startup, so the overloead is minimum

Separate call has much larger diff. Multiple runners has their own main() function (search for FileSystems.setDefaultPipelineOptions) give a hint